.github/workflows/create-production-release.yml

```diff
@@ -1,34 +1,91 @@
-        uses: seanmiddleditch/gha-setup-ninja@master
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.11.0'
-      - uses: ilammy/msvc-dev-cmd@v1
-        with:
-          arch: amd64 # once this is run, gelly-production-release.zip will be created in the root of the repository, and we can just ignore the rest
-      - name: Set up x64 Developer Command Prompt for VS 2022
-        run: |
-          python scripts/make-release.py
-      - name: Upload Release
-        uses: softprops/action-gh-release@v1
-        if: startsWith(github.ref, 'refs/tags/v') # We still want to create a release, but not upload the asset if it's not a tagged release
-        with:
-          files: gelly-production-release.zip
\ No newline at end of file
+    make-gelly-artifacts:
+        runs-on: windows-latest
+        env:
+            SCCACHE_GHA_ENABLED: "on"
+        strategy:
+            matrix:
+                presets: [
+                    { configure: "gelly-gmod-relwithdebinfo", build: "Gelly-GMod-RelWithDebInfo" },
+                    { configure: "gelly-gmod-release", build: "Gelly-GMod-Release" }
+                ]
+        steps:
+            -   name: Checkout latest Gelly
+                uses: actions/checkout@v3
+                with:
+                    path: gelly
+                    submodules: false
+            -   name: Cache Submodules
+                id: cache-submodules
+                uses: actions/cache@v4
+                with:
+                    path: 'gelly/**/vendor/**'
+                    key: gelly-submodules-${{ hashFiles('gelly/.gitmodules') }}
+            -   name: Init all submodules if needed
+                if: steps.cache-submodules.outputs.cache-hit != 'true'
+                run: |
+                    cd gelly
+                    git submodule update --init --recursive
+            -   name: Set up Ninja
+                uses: seanmiddleditch/gha-setup-ninja@master
+            -   uses: ilammy/msvc-dev-cmd@v1
+                name: Set up MSVC
+                with:
+                    arch: amd64
+            -   name: Run sccache-cache
+                uses: mozilla-actions/sccache-action@v0.0.3
+            -   name: Perform build
+                run: |
+                    cd gelly
+                    cmake -DGELLY_PRODUCTION_BUILD=ON --fresh --preset ${{ matrix.presets.configure }} -S . -B bin/${{ matrix.presets.configure }}
+                    cmake --build --target gelly-gmod --preset ${{ matrix.presets.build }}
+            -   name: Add addon to release
+                run: |
+                    cd gelly
+                    scripts/gmad.exe create -folder packages/addon -out release/garrysmod/addons/gelly/gelly.gma
+            -   name: Zip release
+                run: |
+                    cd gelly
+                    Compress-Archive -Path release/* -Destination ${{ matrix.presets.configure }}-x64.zip
```

Comment:
The upload artefact action will automatically compress a directory into a zip if you give it one, making this step redundant